### ISSUE TITLE
CompareMap custom method problems

### ISSUE THREAD

**Issue by RedMser** [ItsDeltin/Overwatch-Script-To-Workshop#96]
Issue 1) CompareMap on a map which doesn't have multiple stages and which is called in a condition block of a rule crashes the compiler. [This part of the source](https://github.com/ItsDeltin/Overwatch-Script-To-Workshop/blob/master/Deltinteger/Deltinteger/Custom%20Methods/CompareMap.cs#L26) seems responsible. The following code

```
rule: "Test"
if (CompareMap(Map.Numbani)) { }
```

outputs

```
An exception was thrown while parsing.
System.InvalidCastException: Unable to cast object of type 'Deltin.Deltinteger.Elements.EnumMember' to type 'Deltin.Deltinteger.Elements.Element'.
   at Deltin.Deltinteger.Parse.TranslateRule.GetConditions(RuleAction ruleAction) in C:\Users\Deltin\Documents\GitHub\Overwatch-Script-To-Workshop\Deltinteger\Deltinteger\Parse\TranslateRule.cs:line 60
   at Deltin.Deltinteger.Parse.TranslateRule..ctor(DeltinScript deltinScript, RuleAction ruleAction) in C:\Users\Deltin\Documents\GitHub\Overwatch-Script-To-Workshop\Deltinteger\Deltinteger\Parse\TranslateRule.cs:line 43
   at Deltin.Deltinteger.Parse.DeltinScript.ToWorkshop(Func`2 addRules) in C:\Users\Deltin\Documents\GitHub\Overwatch-Script-To-Workshop\Deltinteger\Deltinteger\Parse\Translate.cs:line 224
   at Deltin.Deltinteger.LanguageServer.DocumentHandler.Update() in C:\Users\Deltin\Documents\GitHub\Overwatch-Script-To-Workshop\Deltinteger\Deltinteger\Language Server\DocumentHandler.cs:line 179
```

---

Issue 2) CompareMap does not include the Map constant (in both the MapLink branch, and the single-comparison branch), causing Overwatch to fail parsing its output. The following code

```
rule: "Test"
if (CompareMap(Map.Nepal)) {}
```

was tested (but using a regular if-block yields the same result). The compiled condition is

```
conditions
{
	Array Contains(Append To Array(Append To Array(Append To Array(Append To Array(Empty Array, Nepal), Nepal Sanctum), Nepal Shrine), Nepal Village), Current Map) == True;
}
```

but Overwatch expects each map name to be wrapped like `Map(Nepal)`. Error when pasting is:

```
Error: Expected a value after 'Array,' on line 24
```

**Comment by TrueCP6** [comment]
Looks like the first issue can be solved like this where you identified the problem:
```c#
        public override IWorkshopTree Get(ActionSet actionSet, IWorkshopTree[] parameterValues)
        {
            var enumData = parameterValues[0];
            Map map = (Map)((EnumMember)enumData).Value;
            MapLink mapLink = MapLink.GetMapLink(map);

            if (mapLink == null)
                return new V_Compare(new V_CurrentMap(), Operators.Equal, enumData);
            else
                return Element.Part<V_ArrayContains>(mapLink.GetArray(), new V_CurrentMap());
        }
```

**Comment by ItsDeltin** [comment]
Both issues are related, the crash is happening because it is not wrapped in `Map()`.

**Comment by ItsDeltin** [comment]
Thanks for the issue, fixed in [v1.1](https://github.com/ItsDeltin/Overwatch-Script-To-Workshop/releases/tag/v1.1).
